pkg/container/kubernetes/common.go

```diff
@@ -0,0 +1,73 @@
+package kubernetes
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+
+	"github.com/stacklok/toolhive/pkg/logger"
+)
+
+// extra kinds
+const (
+	// defaultRetries is the number of times a resource discovery is retried
+	defaultRetries = 10
+
+	//defaultRetryInterval is the interval to wait before retring a resource discovery
+	defaultRetryInterval = 3 * time.Second
+)
+
+var isOpenShift *bool = nil
```

Comment:
(Not needed to include the PR, we can iterate on the patches in subsequent PRs as well)
I wonder if using sync.Once would be cleaner than using a global variable?

Reply:
Yes, likely so. I had taken the an original implementation mostly verbatim, however I also raised #1341 to track the request.

Reply:
yeah, that's fine. I think the bigger ask here was to make the platform detection extensible and avoid flag-sprawl later if we need to add more platform-specific code later on (what @ChrisJBurns said [here](https://github.com/stacklok/toolhive/pull/1253#discussion_r2261053623))